plugins/event_notification/providers/kafka/lib/model/KafkaNotificationTemplate.php

```diff
@@ -0,0 +1,146 @@
+<?php
+require_once(dirname(__file__) . "/../../../../../../vendor/avro/flix-tech/confluent-schema-registry-api/vendor/autoload.php");
+
+use FlixTech\SchemaRegistryApi\Registry\BlockingRegistry;
+use FlixTech\SchemaRegistryApi\Registry\PromisingRegistry;
+use GuzzleHttp\Client;
+
+/**
+ * @package plugins.kafkaNotification
+ * @subpackage model
+ */
+class KafkaNotificationTemplate extends EventNotificationTemplate
+{
+	const CUSTOM_DATA_TOPIC_NAME = 'topicName';
+	const CUSTOM_DATA_PARTITION_KEY = 'partitionKey';
+	const CUSTOM_DATA_MESSAGE_FORMAT = 'messageFormat';
+	
+	public function __construct()
+	{
+		$this->setType(KafkaNotificationPlugin::getKafkaNotificationTemplateTypeCoreValue(KafkaNotificationTemplateType::KAFKA));
+		parent::__construct();
+	}
+	
+	public function fulfilled(kEventScope $scope)
+	{
+		if (!kCurrentContext::$serializeCallback)
+			return false;
+		
+		if (!parent::fulfilled($scope))
+			return false;
+		
+		return true;
+	}
+	
+	public function setTopicName($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_TOPIC_NAME, $value);
+	}
+	
+	public function getTopicName()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_TOPIC_NAME);
+	}
+	
+	public function setPartitionKey($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_PARTITION_KEY, $value);
+	}
+	
+	public function getPartitionKey()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_PARTITION_KEY);
+	}
+	
+	public function setMessageFormat($value)
+	{
+		return $this->putInCustomData(self::CUSTOM_DATA_MESSAGE_FORMAT, $value);
+	}
+	
+	public function getMessageFormat()
+	{
+		return $this->getFromCustomData(self::CUSTOM_DATA_MESSAGE_FORMAT);
+	}
+	
+	
+	public function dispatch(kScope $scope)
+	{
+		KalturaLog::debug("Dispatching event notification with name [{$this->getName()}] systemName [{$this->getSystemName()}]");
+		if (!$scope || !($scope instanceof kEventScope)) {
+			KalturaLog::err('Failed to dispatch due to incorrect scope [' . $scope . ']');
+			return;
+		}
+		
+		$uniqueId = (string)new UniqueId();
+		$eventTime = date('Y-m-d H:i:s');
+		$eventType = $_REQUEST[action];
+		$objectType = $_REQUEST[service];
```

Comment:
Should probably be getClass of $scope->getEvent()->getObject()

Reply:
$_REQUEST[service]  returns 'baseentry' 
getClass of $scope->getEvent()->getObject()  returns 'entry'
which one is correct?